dbt_sugar/core/task/doc.py

```diff
@@ -148,35 +149,67 @@ def orchestrate_model_documentation(
 
         not_documented_columns = self.get_not_documented_columns(content, model_name)
         self.document_columns(not_documented_columns)
-
-        self.check_tests(schema, model_name)
         self.update_model_description_test_tags(path, model_name, self.column_update_payload)
+        self.check_tests(path, model_name)
+
         # Method to update the descriptions in all the schemas.yml
         self.update_column_descriptions(self.column_update_payload)
-
         return 0
 
-    def check_tests(self, schema: str, model_name: str) -> None:
+    def delete_fail_tests_from_schema(
+        self, path_file: Path, model_name: str, tests_to_delete: Dict[str, List[str]]
+    ):
+        """
+        Method to delete the failing tests from the schema.yml.
+
+        Args:
+            path_file (Path): Path of the schema.yml file to update.
+            model_name (str): Name of the model to document.
+            tests_to_delete (Dict[str, List[str]]): with the tests that have failed.
+        """
+        content = open_yaml(path_file)
+        for model in content["models"]:
+            if model["name"] == model_name:
+                for column in model.get("columns", []):
+                    tests_to_delete_from_column = tests_to_delete.get(column["name"], [])
+                    tests_from_column = column.get("tests", [])
+                    tests_pass = [
+                        x for x in tests_from_column if x not in tests_to_delete_from_column
+                    ]
+                    if not tests_pass and tests_from_column:
+                        del column["tests"]
+                    elif tests_pass:
+                        column["tests"] = tests_pass
+        save_yaml(path_file, content)
+
+    def check_tests(self, path_file: Path, model_name: str) -> None:
         """
         Method to run and add test into a schema.yml, this method will:
 
         Run the tests and if they have been successful it will add them into the schema.yml.
 
         Args:
-            schema (str): Name of the schema where the model lives.
+            path_file (Path): Path of the schema.yml file to update.
             model_name (str): Name of the model to document.
         """
+        dbt_command = f"dbt test --models {model_name}".split()
```

Comment:
I remember reading about this module at some point: https://docs.python.org/3/library/shlex.html#shlex.quote btw to help with command injection.

Do you know if your current method is safe for injections if someone was to pass a test like `rm ~` or something nasty like this?